### DISCUSSION TITLE
Add PNG as an overlay

### DISCUSSION THREAD

**Discussion by radik86** [ffmpeginteropx/FFmpegInteropX#258] [Q&A] [answered]
Hello dear FFmpegInteropX Team,
first of all thank you for your great work!

I have a following question. I try to apply a PNG overlay to my video.
I was trying to put that command  `-stream_loop -1 -i "C:\OneDrive\Imperial\Effects\ImperialOverlayText.png" -filter_complex "[0:v][1:v] overlay=25:25:enable='between(t,0,10)'"` 

with the function  FFmpegMSS?.SetFFmpegVideoFilters()  
Am I able to use stream_loop commands with the f FFmpegMSS?.SetFFmpegVideoFilters() or what would be the propper approach to apply image and video overlay to my main video?

Thank you

Br,
Radik

**Comment by brabebhin** [comment, marked as answer]
If gapless playback isn't your goal, then you can use the template IBasicVideoEffect with win2D to do the overly which has some built in effects for that,  which are faster and easier to apply.

**Reply by lukasf** [reply]
You cannot use input files in filters, due to UWP's sandbox restrictions. We cannot open files by path in UWP. Additionally, using ffmpeg video filters will downgrade you from pure GPU pipeline to a mixed GPU/CPU mode, which will give you bad performance on high res (4K) files.

So as @brabebhin says, you should try to create a Win2D effect. This will be HW accelerated and you can load files using UWP APIs. Take a look at out BasicVideoEffect implementation. You can also get the render time of the image to decide if you need to show the overlay or not, and you can communicate with the effect through the IPropertySet you pass, to change if and what overlay to show.

**Reply by radik86** [reply]
Hi @brabebhin  and @lukasf ,
thank you very much for your quick response and great feedback.

For my purpose the IBasicVideoEffect  should be enough. Thank you for the hint.

Thanks for let me know, that filters are not able to work with additional files input. So I can leverage the lib only for manipulation of the main file.

Thanks again, and have a nice day.

br,
radik

**Reply by brabebhin** [reply]
I always thought the filters might work if the file has been copied inside the app's AppData folder. But never tried it.

**Reply by radik86** [reply]
A further question: are slowmotion and reverse effects possible with your lib?
I cant find any examples of that effects with IBasicVideoEffect and win2D, hence was thinking to do that with ffmpeg. Any suggestions? I would really appreciate it.

Thank you

**Reply by brabebhin** [reply]
The only way to know for sure is if you tried it. I do have some doubts on both of these.

The slow motion probably requires the modification of the video stream descriptors to match the new frame rate. 

The reverse might work.
